packages/medusa/src/api/admin/returns/[id]/receive/route.ts

```diff
@@ -9,12 +16,52 @@ export const POST = async (
   req: AuthenticatedMedusaRequest<AdminPostReceiveReturnsReqSchemaType>,
   res: MedusaResponse
 ) => {
-  const input = req.validatedBody as AdminPostReceiveReturnsReqSchemaType
+  const remoteQuery = req.scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
 
-  const workflow = receiveReturnOrderWorkflow(req.scope)
+  const { id } = req.params
+
+  const workflow = beginReceiveReturnWorkflow(req.scope)
   const { result } = await workflow.run({
-    input,
+    input: {
+      ...req.validatedBody,
+      return_id: id,
+    },
+  })
+
+  const queryObject = remoteQueryObjectFromString({
+    entryPoint: "return",
+    variables: {
+      id: result.return_id,
+      filters: {
+        ...req.filterableFields,
+      },
+    },
+    fields: req.remoteQueryConfig.fields,
   })
 
-  res.status(200).json({ return: result })
+  const [orderReturn] = await remoteQuery(queryObject)
+
+  res.json({
+    order_change: result,
```

Comment:
thought: Would it make sense to return the order + return?

Reply:
yeah, we can do it.
I will change it